temba/utils/gsm7.py

```diff
@@ -377,220 +380,30 @@ def calculate_num_segments(text):
-    "̷": "/",
-    "̸": "/",
-    "⁄": "/",
-    "∕": "/",
-    "⧹": "\\",
-    "⧵": "\\",
-    "": "",
-    "̲": "_",
-    "⃒": "|",
-    "⃓": "|",
-    "∣": "|",
-    "⎸": "|",
-    "⎹": "|",
-    "⏐": "|",
-    "⎜": "|",
-    "⎟": "|",
-    "⎼": "-",
-    "⎽": "-",
-    "―": "-",
-    "‐": "-",
-    "•": "-",
-    "⁃": "-",
-    "ǃ": "!",
-    "‚": "'",
-    "̦": ",",
-    "❨": "(",
-    "❪": "(",
-    "⟮": "(",
-    "⦅": "(",
-    "❩": ")",
-    "❫": ")",
-    "⟯": ")",
-    "⦆": ")",
-    "⁎": "*",
-    "∗": "*",
-    "⊛": "*",
-    "✢": "*",
-    "✣": "*",
-    "✤": "*",
-    "✥": "*",
-    "✱": "*",
-    "✲": "*",
-    "✳": "*",
-    "✺": "*",
-    "✻": "*",
-    "✼": "*",
-    "✽": "*",
-    "❃": "*",
-    "❉": "*",
-    "❊": "*",
-    "❋": "*",
-    "⧆": "*",
-    "˖": "+",
-    "。": ".",
-    "ː": ":",
-    "˸": ":",
-    "⦂": ":",
-    "꞉": ":",
-    "⁏": ";",
-    "͇": "=",
-    "꞊": "=",
-    "ᴀ": "A",
-    "ʙ": "B",
-    "ᴄ": "C",
-    "ᴅ": "D",
-    "ᴇ": "E",
-    "ꜰ": "F",
-    "ɢ": "G",
-    "ʜ": "H",
-    "ɪ": "I",
-    "ᴊ": "J",
-    "ᴋ": "K",
-    "ʟ": "L",
-    "ᴍ": "M",
-    "ɴ": "N",
-    "ᴏ": "O",
-    "ᴘ": "P",
-    "ʀ": "R",
-    "ꜱ": "S",
-    "ᴛ": "T",
-    "ᴜ": "U",
-    "ᴠ": "V",
-    "ᴡ": "W",
-    "ʏ": "Y",
-    "ᴢ": "Z",
-    "ˆ": "^",
-    "̂": "^",
-    "᷍": "^",
-    "❴": "{",
-    "❵": "}",
-    "̃": "~",
-    "˷": "~",
-    "̰": "~",
-    "̴": "~",
-    "∼": "~",
-    "​": "",
-    "̳": "_",
-    "—": "-",
-    "–": "-",
-    "‹": ">",
-    "›": "<",
-    "!!": "!!",
-    "„": '"',
-    "...": "...",
-    "⁠": "",
-    "\t": " ",
-    "\xa0": " ",
-    "¡": "",
-    "£": "",
-    "¤": "",
-    "¥": "",
-    "§": "",
-    "¿": "",
-    "À": "A",
-    "Á": "A",
-    "Â": "A",
-    "Ã": "A",
-    "Ä": "A",
-    "Å": "A",
-    "Æ": "E",
-    "Ç": "C",
-    "È": "E",
-    "É": "E",
-    "Ê": "E",
-    "Ì": "I",
-    "Í": "I",
-    "Î": "I",
-    "Ñ": "N",
-    "Ò": "O",
-    "Ó": "O",
-    "Ô": "O",
-    "Õ": "O",
-    "Ö": "O",
-    "Ø": "O",
-    "Ù": "U",
-    "Ú": "U",
-    "Û": "U",
-    "Ü": "U",
-    "ß": "",
-    "à": "a",
-    "â": "a",
-    "ã": "a",
-    "ä": "a",
-    "å": "a",
-    "æ": "e",
-    "ç": "c",
-    "è": "e",
-    "é": "e",
-    "ê": "e",
-    "ì": "i",
-    "í": "i",
-    "î": "i",
-    "ñ": "n",
-    "ò": "o",
-    "ó": "o",
-    "ô": "o",
-    "õ": "o",
-    "ö": "o",
-    "ø": "o",
-    "ù": "u",
-    "ú": "u",
-    "û": "u",
-    "ü": "u",
-    "Δ": "",
-    "Θ": "O",
-    "Λ": "",
-    "Ξ": "",
-    "Π": "",
-    "Σ": "",
-    "Φ": "",
-    "Ψ": "",
-    "Ω": "",
-}
+def load_gsm7_replacements():
+    handle = open("%s/flow_editor/gsm7-replacements.json" % settings.MEDIA_ROOT, "r+")
+    data = handle.read()
+    handle.close()
+
+    return json.loads(data)
+
+
+GSM_REPLACEMENTS = load_gsm7_replacements()
+
+
+def get_char_replacement(char):
+    key = format(ord(str(char)), "x").zfill(4).upper()
+    replacement = GSM_REPLACEMENTS.get(key, None)
+    print(key, char, replacement)
```

Comment:
If this line was used for debugging I think it needs to be removed. If it is for logging then it is better to use `logger.info`.

Reply:
Thanks for catching that.